tiered-cart-promotions/client/js/lib.js

```diff
@@ -0,0 +1,169 @@
+};
+
+export const getRewardsRender = () => {
+    const renderRewards = rewards => {
+        const rewardsHtmlElement = document.getElementById("reward-wrapper");
+        const rewardsTemplate = document.getElementById("reward-wrapper-template");
+        const rewardItem = rewardsTemplate.cloneNode(true).content;
+        const progressTier = rewardItem.querySelector(".progress-tier");
+        const rect = document.querySelectorAll(".progress-bar-numbers span")[rewards[0]?.hierarchy]?.getBoundingClientRect().left;
+        progressTier.style.width = `${((rect - 20).toFixed(1))}px`;
+        rewardItem.querySelector(".reward-banner").innerHTML = rewards[0]?.hierarchy !== 3 ? rewards[0]?.banner || "Spend $100 more to get FREE SHIPPING" : "Congratulations, you achieved all rewards!";
+        rewardsHtmlElement.replaceChildren(rewardItem);
+        return rewardItem;
+    };
+    return renderRewards;
+};
+
+export const renderProducts = products => {
+    const htmlElement = document.querySelector(".summed-products");
+    const summedProductsHtmlElement = document.querySelector(".summed-products-template");
+    htmlElement.replaceChildren(...products.filter(item => item.quantity).map((item, index) => {
+        const summedProductsTemplate = summedProductsHtmlElement.cloneNode(true).content.children[0];
+        summedProductsTemplate.setAttribute("key", index);
+        summedProductsTemplate.querySelector("img").setAttribute("src", item.src);
+        summedProductsTemplate.querySelector("h6").innerHTML = `${item.productName}`;
+        summedProductsTemplate.querySelector("p").innerHTML = `Quantity ${item.quantity}`;
+        summedProductsTemplate.querySelector("span").innerHTML = `${item.price}`;
+        return summedProductsTemplate;
+    }));
+};
+
+export const renderRewardsFromStorage = (rewards, products) => {
+    const htmlElement = document.querySelector(".discounts");
+    const discountsHtmlElement = document.querySelector(".discounts-template");
+    const discountsTemplate = discountsHtmlElement.cloneNode(true).content;
+    const summedDiscountPrice = rewards?.reduce((sum, voucher) => sum + voucher.discount, 0) / 100;
+    const subtotal = sumProductPrices(products);
+    const discountValue = discountsTemplate.querySelector(".discount-value span").innerHTML = `$${summedDiscountPrice || 0}`;
+    discountsTemplate.querySelector(".all-discounts span").innerHTML = `$${summedDiscountPrice || 0}`;
+    discountsTemplate.querySelector(".subtotal span").innerHTML = `$${subtotal}`;
+    rewards?.map(reward => {
+        discountsTemplate.querySelector(".coupons").innerHTML += `<h5 class="coupon"><span class="coupon-value">Reward Tier ${reward.hierarchy}</span></h5>`;
+    });
+    const shipping = discountsTemplate.querySelector(".shipping span").innerHTML = "$8.99";
+    discountsTemplate.querySelector(".grand-total span").innerHTML = `$${(+shipping.replace("$", "") + +subtotal - discountValue.replace("$", "")).toFixed(2)}`;
```

Comment:
Maybe simplify this case? Assign shipping to variable. '+' stands for parsingInt this values here?

Reply:
yep, + mean that it's a number